book/animations.md

```diff
@@ -1148,25 +1148,53 @@ Compositing algorithms
 ======================
 
 The most complex part of compositing and draw is dealing with the hierarchical
-nature of the display list. To deal with this, you might try this algoritihm:
-mark animating visual effects as composited, and raster a `skia.Surface` for
-everything below it. This works fine for the single `DrawText` example we've
-been looking at, but starts to get very complicated when you consider nested
-surfaces. For example, multiple nested DOM nodes could be simultaneously
-animating opacity, and we can't put the same subtree in two different surfaces.
-Do we instead raster some of it in one surface and some in another? Where do we
-put the combined result with all the opacities applied, or the first applied
-but not yet the second?^[We'll cover another complication---overlap
-testing---later in the chapter; see also the Go Further block at the end of
-this section for discussion of even more complications.]
-
-To handle all this complexity, let's break the problem down into two
-pieces: *compositing* the display list into a linear list of `skia.Surface`s,
-and *drawing* those surfaces to the screen. Compositing is in charge of finding
-non-animating subtrees of the display list and putting them into groups that
-raster together. Drawing  is in charge of re-creating a tree hierarchy that
-mirrors the hierarchy of animating visual effects in the display list.
-[^temp-surface]
+nature of the display list. For example, consider this web page:
+
+``` {.html}
+<div style="opacity:0.999">
+  <p>
```

Comment:
change to span with color

Reply:
Not sure what this is. This bit here is HTML source code; the highlighted bit is lower down.

Reply:
oh it's a <pre> block below that isn't visible to the reader. I withdraw my comment.